packages/compile-common/src/compilerSupplier/strategy.ts

```diff
@@ -0,0 +1,106 @@
+import type { Constructor as FreeConstructor } from "./constructor";
+import type { Results as FreeResults } from "./results";
+
+export namespace Strategy {
+  /**
+   * Type-level description of a compiler supplier strategy
+   */
+  export type Specification = {
+    constructor: Omit<FreeConstructor.Specification, "result">;
+    results: FreeResults.Specification;
+    allowsLoadingSpecificVersion: boolean;
+    allowsListingVersions: boolean;
+  };
+
+  /**
+   * Scoped re-exports of Constructor, using Strategy.Specification as
+   * generic param
+   */
+  export namespace Constructor {
+    /**
+     * Getter for the specified constructor, intersected with fixed result
+     * type of Strategy<S>
+     */
+    export type Specification<S extends Strategy.Specification> =
+      S["constructor"] & { result: Strategy<S>; };
```

Comment:
Why are you adding the `result` field?  I don't actually understand that, it seems to go against what you're mostly doing?

Reply:
In the abstract, a constructor specification must include types for the options as well as types for the instantiated object result... but for a strategy, the constructor's `result` is always going to be that strategy.

Reply:
Oh, I see.  You're restricting that existing field, not adding a new one.

Reply:
Yup